src/control-plane/user-management/user-management.service.ts

```diff
@@ -0,0 +1,72 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import * as apigatewayV2 from 'aws-cdk-lib/aws-apigatewayv2';
+import { HttpLambdaIntegration } from 'aws-cdk-lib/aws-apigatewayv2-integrations';
+import { Construct } from 'constructs';
+import { IAuth } from '..';
+import { IRoute, generateRoutes } from '../../utils';
+
+export interface UserManagementServiceProps {
+  readonly api: apigatewayV2.HttpApi;
```

Comment:
might be good run an LLM through or something to add some jsdocs to some of these parameters

Reply:
Yeah, did this with my latest commit.